src/app.tsx

```diff
@@ -1,29 +1,35 @@
-import reactLogo from './assets/react.svg';
 import { Counter } from 'components/counter';
-import './app.css';
+import { ThemeToggleButton } from 'components/theme-toggle-button';
+import { useThemeStore } from 'features/theming/store';
+import { ConnectedThemeProvider } from 'features/theming/theme-provider';
+import { Card, Heading, Paragraph, Wrapper } from 'styled';
+
+// @TODO(shawk): move this and actually do some routing
+function Router() {
+  const themeStore = useThemeStore();
 
-function App() {
   return (
-    <div className="App">
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src="/vite.svg" className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://reactjs.org" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
+    <Wrapper>
+      <Heading>Vite + React + TypeScript Reference</Heading>
+
+      <Card>
         <Counter />
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </div>
+
+        <Paragraph>
+          Theme: <strong>{themeStore.mode}</strong>
+        </Paragraph>
+
+        <ThemeToggleButton>Toggle theme</ThemeToggleButton>
+      </Card>
+    </Wrapper>
+  );
+}
+
+function App() {
+  return (
+    <ConnectedThemeProvider>
+      <Router />
+    </ConnectedThemeProvider>
```

Comment:
Do you have any thoughts on using standard functions over arrow functions? I tend to prefer arrow functions, but that's just because I started with them.

Reply:
I've used arrow functions a lot in the past, but have been favoring function declarations here.

A few things on my mind:
- specifying generic types on arrow functions sometimes doesn't work properly. This might only be the case in files where `JSX` is being written, since the `<T>` syntax gets misinterpreted
- hoisting doesn't work with arrow functions
- arrow functions are anonymous (not named), so sometimes you get worse info when debugging, though the name is (sometimes?) inferred correctly
- since lexical scope doesn't matter (we don't use `this` in react components), arrow functions aren't really necessary
- the `function` keyword being at the start of the line helps with identifying functions in code (whereas `const fn = ..` starts the same way as any other variable declaration)
- for react components specifically using `function` means you must use `{` .. `}` for the body (no implicit return) which is better in the case where you go from "simple component that just returns JSX" to "needs some variable declarations, logging, debugging, or state"